src/schema/schema_create.c

```diff
@@ -961,6 +977,103 @@ __create_data_source(
     return (dsrc->create(dsrc, &session->iface, uri, (WT_CONFIG_ARG *)cfg));
 }
 
+/*
+ * __import_entry_cmp --
+ *     Qsort function: sort the import entries array by name.
+ */
+static int WT_CDECL
+__import_entry_cmp(const void *a, const void *b)
+{
+    WT_IMPORT_ENTRY *ae, *be;
+
+    ae = (WT_IMPORT_ENTRY *)a;
+    be = (WT_IMPORT_ENTRY *)b;
+
+    return strcmp(ae->name, be->name);
+}
+
+/*
+ * __get_uri_suffix --
+ *     Get suffix of the URI.
+ */
+static int
+__get_uri_suffix(WT_SESSION_IMPL *session, const char *uri, const char **suffix)
+{
+    *suffix = strchr(uri, ':');
+    if (*suffix == NULL || (++(*suffix))[0] == '\0')
+        WT_RET_MSG(session, WT_ERROR, "%s invalid URI", uri);
+
+    return (0);
+}
+
+/*
+ * __metadata_entry_worker --
+ *     Worker function for metadata file reader procedure. The function populates import list with
+ *     entries related to the import URI.
+ */
+static int
+__metadata_entry_worker(WT_SESSION_IMPL *session, WT_ITEM *key, WT_ITEM *value, void *state)
+{
+    WT_IMPORT_LIST *import_list;
+    const char *md_key, *md_key_suffix, *md_value;
+
+    import_list = (WT_IMPORT_LIST *)state;
+    md_key = (const char *)key->data;
+    md_value = (const char *)value->data;
+
+    /* Get suffix of the key. */
+    WT_RET(__get_uri_suffix(session, md_key, &md_key_suffix));
+
+    /* Skip unrelated entries. */
+    if (!WT_PREFIX_MATCH(md_key_suffix, import_list->uri_suffix))
+        return (0);
+
+    /*
+     * We are not checking if the entry already exists in the metadata. It will be handled later in
+     * the appropriate create call.
+     */
+
+    /* Grow the entries array if needed. */
+    WT_RET(__wt_realloc_def(session, &import_list->entries_allocated, import_list->entries_next + 1,
+      &import_list->entries));
+
+    /* Populate the next entry. */
+    WT_RET(__wt_strndup(
+      session, md_key, key->size, &import_list->entries[import_list->entries_next].name));
+    WT_RET(__wt_strndup(
+      session, md_value, value->size, &import_list->entries[import_list->entries_next].config));
+    import_list->entries_next++;
+
+    return (0);
+}
+
+/*
+ * __schema_parse_wt_export --
+ *     Parse export metadata file and populate array of name/config entries related to uri. The
+ *     array is sorted by entry name. Caller is responsible to free any memory allocated for the
+ *     import list.
+ */
+static int
+__schema_parse_wt_export(
+  WT_SESSION_IMPL *session, const char *export_file, WT_IMPORT_LIST *import_list)
+{
+    bool exist;
+
+    exist = false;
+
+    /* Open metadata backup file and iterate over the key value pairs. */
```

Comment:
This comment isn't quite right (it isn't the backup file specifically). I suggest:
Open the specified metadata file and iterate over the key value pairs.

Reply:
Done.